### PR TITLE
Added marketingRegion

### PR DESCRIPTION
**Pull request by willdch**
https://seed-company-squad.monday.com/boards/3451697530/pulses/5186746393

┆Issue is synchronized with this [Monday item](https://seed-company-squad.monday.com/boards/3451697530/pulses/5870648616) by [Unito](https://www.unito.io)

### CODE REVIEW

**Review by CarsonF — CHANGES_REQUESTED** [review]
This should be a computed. It's not stored in the database or updated in a mutation. It's just a helper calculation.

```ts
Project.marketingRegion =>
	(.marketingRegionOverride ?? .marketingCountry.marketingRegion)
```

**Comment by atGit2021** [comment]
> This should be a computed. It's not stored in the database or updated in a mutation. It's just a helper calculation.
> 
> ```ts
> Project.marketingRegion =>
> 	(.marketingRegionOverride ?? .marketingCountry.marketingRegion)
> ```

@CarsonF, do you recommend that we compute the value inside of the hydrate function within the project repository?  I assume we keep the marketingRegion property in the update DTO in order to pass down the value that a user may select to override to, but do we need it in the create DTO?  Let us know what you are thinking.

**Comment by CarsonF** [comment]
> do you recommend that we compute the value inside of the hydrate function within the project repository? 

That sounds like the easiest thing to do.

> I assume we keep the marketingRegion property in the update DTO in order to pass down the value that a user may select to override to, but do we need it in the create DTO?

No the create & update DTOs should not reference this property. It is computed/calculated so where would that user input save to? It doesn't make sense.

File: src/components/project/dto/project.dto.ts [hunk 1]
```diff
@@ -131,6 +131,7 @@ class Project extends Interfaces {
 
   readonly marketingRegionOverride: Secured<IdOf<Location> | null>;
   readonly fieldRegion: Secured<ID | null>;
+  readonly marketingRegion: Secured<IdOf<Location> | null>;
```
Comment (CarsonF) [inline src/components/project/dto/project.dto.ts:134]:
```suggestion
  @Calculated()
  readonly marketingRegion: Secured<IdOf<Location> | null>;
```
This let's the permission system know that edits are not applicable.

Also seen here:
https://github.com/SeedCompany/cord-api-v3/blob/e867d3f50f37fb18c670aef0cd01462b024496c4/src/components/engagement/dto/engagement.dto.ts#L96-L106

**Comment by atGit2021** [comment]
> > do you recommend that we compute the value inside of the hydrate function within the project repository?
> 
> That sounds like the easiest thing to do.
> 
> > I assume we keep the marketingRegion property in the update DTO in order to pass down the value that a user may select to override to, but do we need it in the create DTO?
> 
> No the create & update DTOs should not reference this property. It is computed/calculated so where would that user input save to? It doesn't make sense.

So when a user clicks on the new marketingRegion button in the UI and sets the value to something new, we should use the marketingRegionOverride property to pass that value down to the API and update the override property?

**Comment by CarsonF** [comment]
> So when a user clicks on the new marketingRegion button in the UI and sets the value to something new, we should use the marketingRegionOverride property to pass that value down to the API and update the override property?

Correct.

**Comment by willdch** [comment]
Closing in favor of [PR 3047](https://github.com/SeedCompany/cord-api-v3/pull/3047).

- After creating this branch I realized I needed to have access to the `marketingCountry` property on 3047 in order for the coalesce to work. 
- Added the same changes as a commit on 3047's PR.

**Pull request closed without merging**